9armbot/player.js

```diff
@@ -101,47 +125,46 @@ class Player{
     }
 
     getCoinTop(top = 10){
-        let getPlayers = this.getPlayers("coins", "desc")
-        return this.cloneDeep(getPlayers.splice(0,top>0?top:0))
+        return this.getPlayers("coins", "desc").slice(0, top);
     }
 
-
-    getPlayerByUsername(username){
-        let player = this.players.find(x=>x.username == username)
-        return player || null;
+    getOrCreatePlayer(username) {
+        let player = this.players.find(x => x.username === username);
+        return player || this.create(username);
     }
 
     giveCoins(username, amount){
-        username = username.toLocaleLowerCase();
-        if(!Number(amount)){
-            return
-        }
-        amount = Number(amount)
-        let player = this.players.find(x=>x.username == username)
-        if(player){
-            player.coins+=amount
-        }else{
-            this.create(username,1,amount, "online")
+        let player = this.getOrCreatePlayer(username);
```

Comment:
where is the catch for `amount`? Also some function gives capitalized username. We have to convert `username` to lowercase.

Reply:
Fixed